mcr.cpp

```diff
@@ -1,59 +1,98 @@
+/*
+* @Author: zyg0121 zhouyiguo2012@qq.com
+* @Date: 2024-09-10 01:15:10
+* @LastEditors: zyg0121 zhouyiguo2012@qq.com
+* @LastEditTime: 2024-09-10 01:27:08
+* @FilePath: \week2-mcr1\code.cpp
+* @Description: 
+* 
+* Copyright (c) 2024 by yiguo, All Rights Reserved. 
+*/
+#include <iostream>
+#include <string.h>
+using namespace std;
+
 bool isWin(char game[3][3])
 {
-    bool win = false;
-    if (game[0][0] == game[0][1] && game[0][1] == game[0][2] && (game[0][0] == 'X' || game[0][0] == 'O'))
-        win = true;
-    if (game[1][0] == game[1][1] && game[1][1] == game[1][2] && (game[1][0] == 'X' || game[1][0] == 'O'))
-        win = true;
-    if (game[2][0] == game[2][1] && game[2][1] == game[2][2] && (game[2][0] == 'X' || game[2][0] == 'O'))
-        win = true;
-    // column
-    if (game[0][0] == game[1][0] && game[1][0] == game[2][0] && (game[0][0] == 'X' || game[0][0] == 'O'))
-        win = true;
-    if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] == 'X' || game[0][1] == 'O'))
-        win = true;
-    if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] == 'X' || game[0][2] == 'O'))
-        win = true;
-    // diagonal
-    if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O'))
-        win = true;
-    if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O'))
-        win = true;
-    return win;
+    // Check rows
+    for (int i = 0; i < 3; i++)
+    {
+        if (game[i][0] == game[i][1] && game[i][1] == game[i][2] && (game[i][0] == 'X' || game[i][0] == 'O'))
+            return true;
+    }
+
+    // Check columns
+    for (int j = 0; j < 3; j++)
+    {
+        if (game[0][j] == game[1][j] && game[1][j] == game[2][j] && (game[0][j] == 'X' || game[0][j] == 'O'))
+            return true;
+    }
+
+    // Check diagonals
+    if ((game[0][0] == game[1][1] && game[1][1] == game[2][2]) || (game[0][2] == game[1][1] && game[1][1] == game[2][0]))
+    {
+        if (game[1][1] == 'X' || game[1][1] == 'O')
+            return true;
+    }
+
+    return false;
+}
+
```

Comment:
Add output: print the current status after a player making a step. 

Add tic-tac-toe to the checkerboard state.